lib/internal/queue.js

```diff
@@ -45,16 +47,14 @@ export default function queue(worker, concurrency, payload) {
     }
 
     function _next(tasks) {
-        return function(){
+        return rest(function(args){
             workers -= 1;
 
-            var removed = false;
-            var args = arguments;
             arrayEach(tasks, function (task) {
-                arrayEach(workersList, function (worker, index) {
-                    if (worker === task && !removed) {
+                arraySome(workersList, function (worker, index) {
```

Comment:
I don't think there's any harm in doing it this way -- there's no way this `_next` call can be preempted by another task completing.

Also -- does `arrayEach` support the early return?  e.g. `return false`.